backend/core/src/listings/entities/listing.entity.ts

```diff
@@ -410,6 +412,19 @@ class Listing extends BaseEntity {
     )
   }
 
+  @Expose()
+  @IsOptional({ groups: [ValidationsGroupsEnum.default] })
+  @ApiProperty()
+  get reviewOrderType(): ListingReviewOrder {
+    let lotteryEventExists = false
+    this.events.forEach((event) => {
+      if (event.type === ListingEventType.publicLottery) {
+        lotteryEventExists = true
+      }
+    })
+    return lotteryEventExists ? ListingReviewOrder.lottery : ListingReviewOrder.firstComeFirstServe
+  }
+
```

Comment:
I think this should be moved to the listing DTO. See the other definitions in there with the `@Transform` decorator. Also, with the way it is now generate client doesn't properly add `reviewOrderType` or `ListingReviewOrder`, which causes issues, so the definition should include `reviewOrderType: ListingReviewOrder`.